### ISSUE TITLE
auto-hide/show zone not working properly in newest version

### ISSUE THREAD

**Issue by roman-sc** [piroor/treestyletab#796]
When cursor moves to red zone(screenshots below), tabs will auto-expand.

In old version(0.14.2014051101), its correct, i must touch the tab area to trigger auto-expand.

In new version(0.15.2014111301), the tabs are auto-expanded before moving mouse to tabs - its impossible to click and drag firefox scroll bar. When using Sage addon, the zone is expanded even more - its impossible to click web page links underneath.

ffox 33.0, win7.

correct auto-expand zone in old version 0.14.2014051101
![old](https://cloud.githubusercontent.com/assets/9918410/5159031/b9e09e54-7355-11e4-8ea2-92845fc97cb1.jpg)

incorrect auto-expand zone in new version 0.15.2014111301 default
![new](https://cloud.githubusercontent.com/assets/9918410/5159044/61b6914c-7356-11e4-9b49-a961b991532c.jpg)

incorrect auto-expand zone in new version 0.15.2014111301 with sage addon
![new_sage](https://cloud.githubusercontent.com/assets/9918410/5159046/6dcd2a36-7356-11e4-82f2-9c2f3b99dcfb.jpg)

**Comment by ismirth** [comment]
I am having a similar issue, where once my tab sidebar hides itself I can not hover over the edge to get it to expand, and instead must hold down the ctrl key to have it unhide. This is very frustrating. I have 15 tabs open on one window and I cannot select between them without holding down the ctrl key. Please fix. I have tried making the 'hotzone' larger and that did not effect this problem in the slightest. I could temporarily turn of autohide, but I would rather not do that because I need the screen space.

I have closed all firefox windows and started over from scratch, rebooted, etc, and none of that fixed the issue. Even when a window acts like it will work, if I select a different tab in that window then it stops unhiding after that.

**Comment by eugnsp** [comment]
I suffer from the same problem. Please fix! (Unfortunately the previous version of the addon breaks drag-and-drop functionality of the bookmarks menu in the latest version of FF.)

**Comment by piroor** [comment]
Some changes are introduced around calculation of event coordinates by recent commits. Could you try the latest development build?
http://piro.sakura.ne.jp/xul/xpi/nightly/

**Comment by tsmithe** [comment]
Better: though now scrolling the tab bar with the scroll wheel (on Linux) doesn't work. But the principal issue seems resolved for me.

**Comment by eugnsp** [comment]
Confirm the comment above, except that scrolling with the wheel works on Windows.

**Comment by tsmithe** [comment]
I also have a problem now with opening new windows: the tab bar doesn't automatically disappear after a new window has been opened, wherever I put my mouse cursor. I have to put the window in fullscreen mode for the tab bar to disappear, at which point I can revert to non-fullscreen mode, and all is well.

**Comment by piroor** [comment]
* The problem around auto hide (https://github.com/piroor/treestyletab/issues/796#issuecomment-77230072) has been fixed by recent commits.
 * However, another problem about scrolling (https://github.com/piroor/treestyletab/issues/796#issuecomment-77210810) does not happen on my environment. Because it is off-topic, it should be posted as a separate issue.

So I think I can close this issue.

**Comment by tsmithe** [comment]
It all seems to work for me, now -- thanks!

**Comment by eugnsp** [comment]
After one day of testing, some problem seems to be still there, though not as serious as it was. I have the tree tabs panel on the left of the screen, and it does not show up, when I quickly move the mouse cursor to the left edge of the screen. When I move it slowly, it works fine. Moreover, auto show does not work at all when PDF file is displayed via PDF plugins (I've checked Foxit Reader and Adobe Reader plugins).

**Comment by tsmithe** [comment]
Actually, yes, I can confirm that rapid movement of the cursor to the edge where the tab bar is hiding does not reveal the tab bar. There seems to be a few-pixels-wide border there -- as soon as my cursor moves inside of that border, the tab bar is revealed. But since the border is part of the Firefox window, I would expect it to be sensitive to the mouse cursor, too.

**Comment by roman-sc** [comment]
Hi, i can confirm that my initial problem was fixed(from my first post).
Im testing latest dev version right now. Like guys above me, im having the same problem.
Sometimes, the tab bar does not auto-expand, nomatter what, even when i move the cursor inside minimalized area. I have to move cursour out of the tab bar and back inside to expand it.
This happened already two times in last 5minutes of my testing, so i guess its pretty common?

**Comment by stth1998** [comment]
Hi,

for me, this misplaced hover zone appeared to be a feature on my linux machine. I have the Tab bar on the left hand side and the close buttons on the right hand side of the tab bar. So, I can hit the close buttons without expanding the tab bar and re-aiming at the cloas ebutton. This is a great feature for me. Sadly, this does not work in firefox on windows. Should I open a separate feature req for this?

Cheers, stth

**Comment by piroor** [comment]
Does this still happen with the latest development build?
http://piro.sakura.ne.jp/xul/xpi/nightly/

Note: if you cannot install the development build with "unsigned" error, you'll have to disable the signature check. Go to `about:config` and turn `xpinstall.signatures.required` to `false`. The preference won't available on released versions of Firefox 44 and later, so you'll have to use Firefox Developer Edition or Nightly to try the development build of TST.

**Comment by roman-sc** [comment]
I have updated my firefox(by accident :( ) after long time. Your extension was not working so i had to update it. On newest production/nightly versions, all my previous problems with auto-expand have been solved.
Unfortunately there is brand new and very anoying problem. I made a video so its easy to see.
https://vid.me/megd
When clicking on or from active or reloaded tab, the whole treestyletab section shrinks and then expands when i move the cursor again. At the start of the video, im trying to click on new tabs. When my cursor is still, im using my mouse scroll wheel. Whole treestyletab keeps expanding and shrinking without my cursor leaving the tab zone. It seems to be happening when some tab updates(such as gmail) or is active and playing youtube video or you reload the page or click new link etc.
My primary ffox version is 43.0.1 on windows. To make sure that this is not connected to my old firefox/other addons, i have downloaded ffox standalone 43.0.4 portable and installed your addon but i had the same problem even there.
